lib/Dialect/FIRRTL/Transforms/LowerClasses.cpp

```diff
@@ -1857,14 +1857,16 @@ static void populateTypeConverter(TypeConverter &converter) {
   converter.addTargetMaterialization(
       [](OpBuilder &builder, Type type, ValueRange values, Location loc) {
         assert(values.size() == 1);
-        return values[0];
+        return builder.create<UnrealizedConversionCastOp>(loc, type, values[0])
```

Comment:
The comment above this says the materialization here is to fold away the casts, which seems unlikely to be what is accomplished by the new version.

Should this just be dropped? (And the other here?)
(I can help answer that later)

Thanks for helping maintain this!

Reply:
Right, it is also not what is accomplished by the old version. Let me change that comment to make clear what it does.